docker/Makefile

```
@@ -142,18 +142,12 @@ push:
 
 # The tests are written in Python. Make a virtualenv to handle the dependencies.
 venv: requirements.txt
-	@if [ -z $$PYTHON3 ]; then\
-	    PY3_MINOR_VER=`python3 --version 2>&1 | cut -d " " -f 2 | cut -d "." -f 2`;\
-	    if (( $$PY3_MINOR_VER < 5 )); then\
-		echo "Couldn't find python3 in \$PATH that is >=3.5";\
-		echo "Please install python3.5 or later or explicity define the python3 executable name with \$PYTHON3";\
-		echo "Exiting here";\
-		exit 1;\
-	    else\
-		export PYTHON3="python3.$$PY3_MINOR_VER";\
-	   fi;\
-	fi;\
-	test -d venv || virtualenv --python=$$PYTHON3 venv;\
+	LOCAL_PY_VER=`python3 --version 2>&1`;\
+	echo "Was using $$LOCAL_PY_VER";\
+	eval "$$(pyenv init -)" && eval "$$(pyenv init --path)";\
+	pyenv install --skip-existing $(PY_VERSION);\
+	pyenv local $(PY_VERSION);\
+	python3 -mvenv venv; \
 	pip install -r requirements.txt;\
 	touch venv;\
```

Let's ensure that if any of these commands fails we terminate. With the existing ";", and the last instruction being a touch (that typically doesn't fail), it's likely that silent failures here cause failures elsewhere later.

```suggestion
	LOCAL_PY_VER=`python3 --version 2>&1`&&\
	echo "Was using $$LOCAL_PY_VER" &&\
	eval "$$(pyenv init -)" && eval "$$(pyenv init --path)" &&\
	pyenv install --skip-existing $(PY_VERSION) &&\
	pyenv local $(PY_VERSION) &&\
	python3 -mvenv venv && \
	pip install -r requirements.txt &&\
	touch venv &&\
```

---

Example of local env without pyenv:

```
❯ make venv
LOCAL_PY_VER=`python3 --version 2>&1`;\
	echo "Was using $LOCAL_PY_VER";\
	eval "$(pyenv init -)" && eval "$(pyenv init --path)";\
	pyenv install --skip-existing 3.6.13;\
	pyenv local 3.6.13;\
	python3 -mvenv venv; \
	pip install -r requirements.txt;\
	touch venv;\

Was using Python 3.9.7
/bin/bash: pyenv: command not found
/bin/bash: pyenv: command not found
/bin/bash: pyenv: command not found
/bin/bash: pyenv: command not found
Collecting flake8==3.4.1
  Using cached flake8-3.4.1-py2.py3-none-any.whl (68 kB)
Collecting jinja2-cli[yaml]==0.7.0
  Using cached jinja2_cli-0.7.0-py2.py3-none-any.whl (6.2 kB)
Collecting jinja2==2.11.3
  Using cached Jinja2-2.11.3-py2.py3-none-any.whl (125 kB)
Collecting retrying==1.3.3
[...]
```